src/Microsoft.SqlTools.ServiceLayer/Scripting/ScriptingService.cs

```diff
@@ -155,10 +151,34 @@ public void InitializeService(ServiceHost serviceHost)
             }
         }
 
-        /// <summary>
-        /// Handles request to start the scripting operation
-        /// </summary>
-        public async Task HandleScriptExecuteRequest(ScriptingParams parameters, RequestContext<ScriptingResult> requestContext)
+        private void RunScriptAsTask(ConnectionInfo connInfo, ScriptingParams parameters, RequestContext<ScriptingResult> requestContext)
+        {
+            ScriptAsScriptingOperation operation = new ScriptAsScriptingOperation(parameters);
+            ConnectionServiceInstance.ConnectionQueue.QueueBindingOperation(
+                key: ConnectionServiceInstance.ConnectionQueue.AddConnectionContext(connInfo, "Scripting"),
+                bindingTimeout: ScriptingOperationTimeout,
+                bindOperation: (bindingContext, cancelToken) =>
+                {
+                    string script = string.Empty;
+                    operation.ServerConnection = bindingContext.ServerConnection;
```

Comment:
this is a bit of a problem in that we are taking values from bindingContext and saving them to a variable that survives outside the locked action section.  this isn't really a good design practice as this context should only be used within the locked callback functions.